### PR TITLE
Refactor generate_connectors to use entities

### PR DESCRIPTION
**Pull request by rnestler**

### CODE REVIEW

**Comment by rnestler** [comment]
@dbrgn Ready for review.

**Comment by rnestler** [comment]
@ubruhin This introduces a small change in some of the symbols, it changes the the with of the symbol outlines:
```diff
-  (width 0.1875) (fill false) (grab_area false)
+  (width 0.188) (fill false) (grab_area false)
```

Should I bump the version for this change?

**Comment by ubruhin** [comment]
> Should I bump the version for this change?

Yes, I would bump either the patch or the minor version number.

**Comment by dbrgn** [comment]
@ubruhin note that the only difference is that we now consistently round to 3 digits after the decimal point, so strictly speaking you won't notice any difference. I don't think the version bump would be required in that case, but it doesn't do any damage either.

File: generate_connectors.py [hunk 1]
```diff
@@ -275,33 +275,29 @@ def _uuid(identifier):
...
+        symbol = Symbol(uuid_sym,
+                        Name('{} 1x{}'.format(name, i)),
+                        Description('A 1x{} {}.\\n\\n'
+                                    'Generated with {}'.format(i, name_lower, generator)),
+                        Keywords('connector, 1x{}, {}'.format(i, keywords)),
+                        Author(author),
+                        Version(version),
+                        Created(create_date or now()),
+                        Category(cmpcat),
+                        )
```
Comment (dbrgn) [inline generate_connectors.py]: Nitpick: PEP8 would disagree with visual alignment :slightly_smiling_face:

File: generate_connectors.py [hunk 2]
```diff
@@ -310,53 +306,39 @@ def _uuid(identifier):
...
-            ff(sym_text_height), ff(y_max),
-        ))
-        lines.append(' )')
-        lines.append(' (text {} (layer sym_values) (value "{{{{VALUE}}}}")'.format(uuid_text_value))
-        lines.append('  (align center top) (height {}) (position 0.0 {}) (rotation 0.0)'.format(
-            ff(sym_text_height), ff(y_min),
-        ))
-        lines.append(' )')
+        text = Text(uuid_text_name, Layer('sym_names'), Value('{{NAME}}'), Align('center bottom'), Height(sym_text_height), Position(0.0, y_max), Rotation(0.0))
+        symbol.add_text(text)
```
Comment (rnestler) [inline generate_connectors.py]: We could add some dedicated helpers like `add_name` and `add_value` which use the conventions for align, position and rotation.

**Pull request merged**